openedx_lti_tool_plugin/views.py

```diff
@@ -51,6 +57,42 @@ def wrapped_view(*args, **kwargs):
 class LtiBaseView(View):
     """Base LTI view initializing common attributes."""
 
+    def check_user_is_enrolled(self, edx_user: UserT, course_id: str):
+        """
+        Check user is enrolled to course.
+
+        Args:
+            edx_user:  Open edX user instance.
+            course_id: course_id: Course ID string.
+
+        Raises:
+            Http404: Course enrollment is not found.
+        """
+        if not course_enrollment().get_enrollment(edx_user, CourseKey.from_string(course_id)):
+            raise Http404
+
+    def get_course_outline(self, request: HttpRequest, course_id: str) -> Union[dict, HttpResponseBadRequest]:
```

Comment:
I think this method must be available only in the CourseHome view. @kuipumu

Reply:
@Squirrel18 This method is also needed on the courseware view to render the course outline.